lib/util/LocationStatusManager.js

```diff
@@ -52,18 +131,262 @@ class LocationStatusManager {
             .filter(svc => serviceConfig[svc].isMongo);
 
         this._locationStatusColl = null;
-
         this._locationStatusStore = {};
         this._scheduledResumeJobs = {
             crr: {},
             ingestion: {},
             lifecycle: {},
         };
+
+        this._serviceLocationStates = new Map();
```

Comment:
we already store the state (for each workflow/location) in `LocationStatus` instances, in `_locationStatusStore` : so the state is kind of redundant....